File: client/src/main/java/com/jraska/github/client/ui/ErrorModel.java
package com.jraska.github.client.ui;

import android.support.annotation.DrawableRes;
import android.view.View;
import android.widget.ImageView;

import com.airbnb.epoxy.EpoxyModel;
import com.jraska.github.client.R;

public final class ErrorModel extends EpoxyModel<View> {
  @DrawableRes
  private final int icon;

  public ErrorModel(int icon) {
    this.icon = icon;
  }

  @Override protected int getDefaultLayout() {
    return R.layout.item_generic_error;
  }

  @Override public void bind(View view) {
    ((ImageView) view.findViewById(R.id.image)).setImageResource(icon);
  }

  static ErrorModel networkError() {
    return new ErrorModel(R.drawable.ic_signal_wifi_off_black_48dp);
  }

  static ErrorModel genericError() {
    return new ErrorModel(R.drawable.ic_error_outline_black_48dp);
  }
}

File: feature-users/src/main/java/com/jraska/github/client/users/ui/ErrorModel.kt
package com.jraska.github.client.users.ui

import android.view.View
import android.widget.ImageView
import androidx.annotation.DrawableRes
import com.airbnb.epoxy.EpoxyModel
import com.jraska.github.client.users.R

class ErrorModel(@DrawableRes private val icon: Int) : EpoxyModel<View>() {

  override fun getDefaultLayout(): Int {
    return R.layout.item_generic_error
  }

  override fun bind(view: View) {
    (view.findViewById<View>(R.id.image) as ImageView).setImageResource(icon)
  }

  companion object {

    internal fun networkError(): ErrorModel {
      return ErrorModel(R.drawable.ic_signal_wifi_off_black_48dp)
    }

    internal fun genericError(): ErrorModel {
      return ErrorModel(R.drawable.ic_error_outline_black_48dp)
    }
  }
}
